sphinx/config.py

```diff
@@ -257,6 +257,9 @@ class Config(object):
         linkcheck_timeout = (None, None, [int]),
         linkcheck_workers = (5, None),
         linkcheck_anchors = (True, None),
+        # Anchors starting with /! are ignored since they are
+        # commonly used for dynamic pages
+        linkcheck_anchors_ignore = (["/#!"], None),
```

Comment:
This is a slight behaviour change in that it requires URI to end in a `/`, no?
Previously, `http://example.com/foo#!bar` would be ignored whereas now it won't.

Also, good indication this is lacking a test. :-)